File: src/main/java/org/smarthata/service/DateUtils.java
package org.smarthata.service;

import java.time.Duration;
import java.time.LocalDateTime;

import static java.time.LocalDateTime.now;

public class DateUtils {
    public static boolean isDateAfter(LocalDateTime time, Duration duration) {
        return time == null || Duration.between(time, now()).compareTo(duration) > 0;
    }
}

File: src/main/kotlin/org/smarthata/service/DateUtils.kt
package org.smarthata.service

import java.time.Duration
import java.time.LocalDateTime

object DateUtils {
    @JvmStatic
    fun isDateAfter(time: LocalDateTime?, duration: Duration): Boolean {
        return time == null || Duration.between(time, LocalDateTime.now()) > duration
    }
}